### PR TITLE
[5863] Remove specialist teaching subject

### PR DESCRIPTION
**Pull request by darokel**
### Context

For 2023 to 2024 we are removing the concept of Specialist teaching from Register. This generally means Primary with mathematics.

In our UI, we already hide this subject from users. When users add a trainee with `Primary with mathematics`, previously, behind the scenes we set `course_subject_one` to `specialist teaching (primary with mathematics)`. This PR undoes all of that.

### Changes proposed in this pull request

- Removes the constant for `specialist teaching (primary with mathematics)`
- Removes the `specialist teaching (primary with mathematics)` allocation subject 
- Trainees added with `Primary with mathematics` get  two subjects set `primary` and `mathematics` (no different than other primary subject combos

### Guidance to review

- Create a new trainee, choose `Primary with mathematics` and assert that the trainee has two different subjects set `primary` and `mathematics`

### CODE REVIEW

**Comment by darokel** [comment]
[Remove 'specialist teaching' from Register](https://trello.com/c/J3SYFoXc/5863-remove-specialist-teaching-from-register)

File: .gitignore [hunk 1]
```diff
@@ -75,3 +75,4 @@ terraform/aks/vendor/
 Procfile.dev
 missing-location-graph-from-dttp.json
 .vscode/*
+Caddyfile
```
Comment (darokel) [inline .gitignore:78]: We used this in Publish as a reverse proxy for local dev with SSL. Just adding this here for consistency as I use it to do the same for Register

**Review by carlosmartinez — APPROVED** [review]
so much delete, so happy

**Comment by darokel** [comment]
[SPIKE: PUBLISH_PRIMARY_SUBJECT_SPECIALISM_MAPPING error](https://trello.com/c/IWsRglxk/5954-spike-publishprimarysubjectspecialismmapping-error)

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**